src/main/java/Main.java

```diff
@@ -8,6 +10,78 @@ public static void main(String[] args) {
     }
 
     public void run(Scanner scanner) {
-        // napisz swój program tutaj. Do wczytywania danych użyj przekazanego w parametrze scannera
+        List<Integer> numbers = createList(scanner);
+
+        reverseAndPrintNumbers(numbers);
+        System.out.println();
+        sumAndPrintNumbers(numbers);
+        System.out.println("Najmniejsza liczba w liście to " + findLowestNumber(numbers));
+        System.out.println("Największa liczba w liście to " + findHighestNumber(numbers));
+    }
+
+    private List createList(Scanner scanner) {
+        List<Integer> numbers = new ArrayList<>();
+        System.out.println("Podaj liczby");
+        int userInput = scanner.nextInt();
+
+        try {
+            if (userInput <= 0) {
+                throw new IncorrectNumberException();
+            }
+        } catch (IncorrectNumberException e) {
+            System.err.println(e.getMessage());
+        }
```

Comment:
Ten fragment nic Ci nie daje i można go usunąć. Sprawdzanie, czy liczba jest większa od 0 masz już w while i to wystarczy.

Dodatkowo używanie wyjątków tutaj nie jest wskazane, bo to samo można by zapisać po prostu jako
```
if (userInput <= 0) {
    System.err.println(jakiś komunikat);
}
```